chromadb/db/mixins/sysdb.py

```diff
@@ -451,6 +454,12 @@ def get_collections(
                     )
                 )
 
+            # apply limit and offset
+            if limit is not None:
```

Comment:
Note: this is inefficient at larger N

Reply:
definitely - do you think we should bite that complexity now or leave it for later?

Reply:
@jeffchuber, check my impl above; I've tested it and it seems to work.

Reply:
this?

```py
collections = list(itertools.islice(self._collections(cur,q),
                                                offset,
                                                min(offset + limit, sys.maxsize)))
```

this still seems to fetch all and then slice down?

Reply:
nope, it fetches slices up to the one we need though, which is still a slight performance gain over fetching the full thing. It's taken from how we paginate documents.

Reply:
you are referencing this code above right? https://github.com/chroma-core/chroma/pull/1451#discussion_r1419055983

im confused because limit and offset never make to to the query level... so how could they be applied there?

Reply:
> definitely - do you think we should bite that complexity now or leave it for later?

Up to you. Just noting.

Reply:
@jeffchuber:

```py
cur_iterator = iter(cur.fetchone, None)
```

We use an iterator to fetch individual records, and we stop at the limit + offset.